spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/source/SparkScan.java

```diff
@@ -80,10 +80,18 @@ abstract class SparkScan implements Scan, SupportsReportStatistics {
     this.branch = readConf.branch();
   }
 
+  protected JavaSparkContext sparkContext() {
```

Comment:
Can we get the parallelism a different way? What about exposing just the parallelism and not actually the conf or context?

Reply:
I moved the method to `SparkScan` and neither `sparkContext` nor `readConf` are exposed now.